### PR TITLE
Remove JNDI support

### PR DESCRIPTION
**Pull request by basil**
This code dates back to 2008. I think it's safe to say that in 2021 JNDI support is not desirable for a project like Jenkins.

### Proposed changelog entries

Remove support for setting the Jenkins home directory via Java Naming and Directory Interface (JNDI).

### Proposed upgrade guidelines

Support for setting the Jenkins home directory via Java Naming and Directory Interface (JNDI) has been removed. The Jenkins home directory may still be set using Java system properties or OS environment variables.

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
I am +1 for the change itself. I guess nobody heard about JNDI for years before log4shell. I suggest mentioning it in the upgrade guide just in case though

**Review by StefanSpieker — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by gmshake** [comment]
@basil
This is a breaking change.
I have an instance running with JNDI JENKINS_HOME out of box for years and it stops working after upgrading to 2.332.1.  Sadly the LTS upgrade guide does not mention it.

**Comment by awillinger-work** [comment]
This sadly also broke our installation... we are running two Jenkins instances on one Wildfly server, where each instance has its own home directory set through the web.xml. After starting any instance with the new version, we are greeted with the initial setup page, as the web.xml ist ignored because of this change. Now we have to split them up into different Wildfly instances...

On another note: The web.xml still contains the parameter HUDSON_HOME which is empty by default. Considering this change caused web.xml env vars to be ignored now, maybe it should be removed as well? Shall I create a JIRA ticket?

**Comment by basil** [comment]
> Shall I create a JIRA ticket?

I opened #6111, but for future reference, a pull request from you would have been appreciated.

**Comment by johnou** [comment]
@basil this change broke our Jenkins installation as well, we are running two Jenkins web applications in a single Tomcat container which pulled the JENKINS_HOME from webapp specific environment variables using JNDI, using Java system properties is not possible since you can only have one per JVM..

**Comment by basil** [comment]
> we are running two Jenkins web applications in a single Tomcat

This is just no longer supported. Sorry!

**Pull request merged**
